be/src/runtime/runtime_filter_worker.cpp

```diff
@@ -509,10 +556,58 @@ void RuntimeFilterWorker::_receive_total_runtime_filter(PTransmitRuntimeFilterPa
         }
 
         index += (1 + half);
+        _exec_env->add_rf_event(request.query_id(), request.filter_id(), addr.hostname, "FORWARD");
         send_rpc_runtime_filter(stub, rpc_closure, default_send_rpc_runtime_filter_timeout_ms, request);
     }
 }
 
+void RuntimeFilterWorker::_send_broadcast_runtime_filter(PTransmitRuntimeFilterParams&& params,
+                                                         std::vector<TRuntimeFilterDestination>&& destinations,
+                                                         int timeout_ms) {
+    DCHECK(!destinations.empty());
+    std::random_device rd;
+    std::mt19937 rand(rd());
+    std::shuffle(destinations.begin(), destinations.end(), rand);
+    _exec_env->add_rf_event(params.query_id(), params.filter_id(),
+                            strings::Substitute("SEND_BROADCAST_RF_RPC: num_dest=$0", destinations.size()));
+    params.set_is_partial(false);
+    TNetworkAddress local;
+    local.hostname = BackendOptions::get_localhost();
+    local.port = config::brpc_port;
+    for (auto i = 1; i < destinations.size(); ++i) {
+        if (destinations[i].address == local) {
+            std::swap(destinations[i], destinations[0]);
+            break;
+        }
+    }
+
+    std::vector<RuntimeFilterRpcClosure*> rpc_closures(destinations.size());
+
+    for (int i = 0; i < destinations.size(); ++i) {
```

Comment:
every destinations has different hostname?

Reply:
yes, FE group destinations by BE.